File: tests/scripts/linear_regression_arg.py
import sys
import pandas as pd
import statsmodels.api as sm

# --- Load input files from command line ---
feature_file = sys.argv[1]
pheno_file = sys.argv[2]

# --- Load features ---
features = pd.read_csv(feature_file, sep='\t')
features = features.rename(columns={features.columns[0]: "IID"})

# --- Load phenotype ---
pheno = pd.read_csv(pheno_file, sep='\t', usecols=["IID", "PHENO"])

# --- Merge on sample ID ---
df = pd.merge(features, pheno, on="IID")

# --- Prepare data for model ---
X = df.drop(columns=["IID", "PHENO"])

# Add intercept
X = sm.add_constant(X)

# Target variable
y = df["PHENO"]

print("X : ", X)
print("y : ", y)

# Fit OLS model
model = sm.OLS(y, X).fit()

# Print full summary
print(model.summary())

# python3 linear_regression.py ../output/regression/48_51.tsv ../data/quantitative/phenotype.tsv

File: tests/regression/linear_regression_arg.R
# --- Load libraries ---
args <- commandArgs(trailingOnly = TRUE)
library(readr)
library(dplyr)

# --- Load input files from command line ---
feature_file <- args[1]
pheno_file <- args[2]

# --- Load features ---
features <- read_tsv(feature_file)
colnames(features)[1] <- "IID"

# --- Load phenotype ---
pheno <- read_tsv(pheno_file, col_types = cols_only(IID = col_character(), PHENO = col_double()))

# --- Merge on sample ID ---
df <- inner_join(features, pheno, by = "IID")

# --- Prepare data for model ---
X <- df %>% select(-IID, -PHENO)
y <- df$PHENO

# --- Add intercept manually (R does this by default in lm) ---
model <- lm(y ~ ., data = X)

# --- Print summary ---
summary(model)

# Run it in terminal like:
# remove the last column header if column empty
# Rscript linear_regression.R ../output/regression/48_51.tsv ../data/quantitative/phenotype.tsv
